pkg/controller/keycloak/keycloak_migrations.go

```diff
@@ -42,10 +42,18 @@ func needsMigration(cr *v1alpha1.Keycloak, currentState *common.ClusterState) bo
 	if currentState.KeycloakDeployment == nil {
 		return false
 	}
-	deployedImage := currentState.KeycloakDeployment.Spec.Template.Spec.Containers[0].Image
-	currentImage := model.KeycloakImage
+
+	currentImage := currentState.KeycloakDeployment.Spec.Template.Spec.Containers[0].Image
+	deployedImage := model.KeycloakImage
+	if cr.Spec.ImageOverrides.Keycloak != "" {
+		deployedImage = cr.Spec.ImageOverrides.Keycloak
+	}
+
 	if cr.Spec.Profile == common.RHSSOProfile {
-		currentImage = model.RHSSOImage
+		deployedImage = model.RHSSOImage
+		if cr.Spec.ImageOverrides.RHSSO != "" {
+			deployedImage = cr.Spec.ImageOverrides.RHSSO
+		}
 	}
```

Comment:
Hi @slaskawi ,

I made some change here after your first review.
Please review this place.